applications/GeoMechanicsApplication/python_scripts/geomechanics_analysis.py

```diff
@@ -218,6 +218,11 @@ def RunSolutionLoop(self):
             self.FinalizeSolutionStep()
             self.OutputSolutionStep()
 
+# For orchestrator-based multi-stage analyses, we need to register our own analysis stage class
+Kratos.Registry.AddItem("Stages.KratosMultiphysics.GeoMechanicsApplication.geomechanics_analysis.GeoMechanicsAnalysis.ClassName", "GeoMechanicsAnalysis")
+Kratos.Registry.AddItem("Stages.KratosMultiphysics.GeoMechanicsApplication.geomechanics_analysis.GeoMechanicsAnalysis.ModuleName", "KratosMultiphysics.GeoMechanicsApplication.geomechanics_analysis")
```

Comment:
This should be placed in the registry lists to happen when importing the application module. Note that current implementation implies that the happens when importing the stage module, something that precludes using the registry as a factory.

See an example in the `FluidDynamicsApplication` [here](https://github.com/KratosMultiphysics/Kratos/blob/6cca9bb7cc18f7c30bb800d89952acd407b6ed93/applications/FluidDynamicsApplication/python_registry_lists.py) and [here](https://github.com/KratosMultiphysics/Kratos/blob/6cca9bb7cc18f7c30bb800d89952acd407b6ed93/applications/FluidDynamicsApplication/FluidDynamicsApplication.py).